src/renderers/shared/stack/reconciler/ReactRef.js

```diff
@@ -56,20 +66,21 @@ ReactRef.shouldUpdateRefs = function(prevElement, nextElement) {
   // is made. It probably belongs where the key checking and
   // instantiateReactComponent is done.
 
-  var prevEmpty = prevElement === null || prevElement === false;
-  var nextEmpty = nextElement === null || nextElement === false;
-
   return (
     // This has a few false positives w/r/t empty components.
-    prevEmpty || nextEmpty ||
+    prevElement == null || prevElement === false || // prevEmpty
```

Comment:
> It would be nice if we could enforce this somehow but I guess making the check more relaxed should not be a big deal.

Flow is really good at describing what types a value can be and enforcing it. Instead of having everyone aggressively check things even though they couldn't possibly be in that state but it's too hard to figure it out by reading at the code.